graph-proxy/src/graphql/workflows.rs

```diff
@@ -27,175 +28,221 @@ pub(super) enum WorkflowParsingError {
-        match value.phase.as_deref() {
-            Some("Pending") => Ok(Some(Self::Pending(WorkflowPendingStatus::from(value)))),
-            Some("Running") => Ok(Some(Self::Running(WorkflowRunningStatus::new(
-                value, metadata,
-            )?))),
-            Some("Succeeded") => Ok(Some(Self::Succeeded(
-                WorkflowCompleteStatus::new(value, metadata)?.into(),
-            ))),
-            Some("Failed") => Ok(Some(Self::Failed(
-                WorkflowCompleteStatus::new(value, metadata)?.into(),
-            ))),
-            Some("Error") => Ok(Some(Self::Errored(
-                WorkflowCompleteStatus::new(value, metadata)?.into(),
-            ))),
-            Some(_) => Err(WorkflowParsingError::UnrecognisedPhase),
-            None => Ok(None),
+        match workflow.status.as_ref() {
+            Some(status) => match status.phase.as_deref() {
+                Some("Pending") => Ok(Some(Self::Pending(WorkflowPendingStatus(status)))),
+                Some("Running") => Ok(Some(Self::Running(WorkflowRunningStatus {
+                    manifest: status,
+                    metadata,
+                }))),
+                Some("Succeeded") => Ok(Some(Self::Succeeded(
+                    WorkflowCompleteStatus {
+                        manifest: status,
+                        metadata,
+                    }
+                    .into(),
+                ))),
+                Some("Failed") => Ok(Some(Self::Failed(
+                    WorkflowCompleteStatus {
+                        manifest: status,
+                        metadata,
+                    }
+                    .into(),
+                ))),
+                Some("Error") => Ok(Some(Self::Errored(
+                    WorkflowCompleteStatus {
+                        manifest: status,
+                        metadata,
+                    }
+                    .into(),
+                ))),
+                Some(_) => Err(WorkflowParsingError::UnrecognisedPhase),
+                None => Ok(None),
+            },
+            None => Err(WorkflowParsingError::MissingWorkflowStatus),
         }
     }
 }
 
 /// No tasks within the workflow have been scheduled
-#[derive(Debug, SimpleObject)]
-struct WorkflowPendingStatus {
-    /// A human readable message indicating details about why the workflow is in this condition
-    message: Option<String>,
-}
+#[derive(Debug)]
+struct WorkflowPendingStatus<'a>(&'a IoArgoprojWorkflowV1alpha1WorkflowStatus);
 
-impl From<IoArgoprojWorkflowV1alpha1WorkflowStatus> for WorkflowPendingStatus {
-    fn from(value: IoArgoprojWorkflowV1alpha1WorkflowStatus) -> Self {
-        Self {
-            message: value.message,
-        }
+#[Object]
+impl WorkflowPendingStatus<'_> {
+    /// A human readable message indicating details about why the workflow is in this condition
+    async fn message(&self) -> Option<String> {
+        self.0.message.clone()
     }
```

Comment:
Instead of cloning the message we could pass a ref
```suggestion
    async fn message(&self) -> Option<&str> {
        self.0.message.as_ref()
    }
```

Reply:
Good catch, I do the same thing in a few other workflow status structs